### PR TITLE
Added exponetiation by squaring algorithm in javascript

### PR DESCRIPTION
**Pull request by devblin**
**Hi!**  
I have added exponentiation by squaring algorithm in javascript.
---
Fixes #617 
By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-expsq/javascript/exponentiation_by_squaring.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		power >>= 1; //right-shift to make power half
+	}
+	return result;
+}
+
+function main() {
+	let power = 2; 
+	let base = 6;
+
+	console.log(exponentiationSquaring(base, power));
```
Comment (RavinduSachintha) [inline algorithms/ar-expsq/javascript/exponentiation_by_squaring.js:17]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6351589)

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@devblin would you be able to add this contribution to the algorithm chart on readme file? don't change anything else.

**Comment by devblin** [comment]
Yeah, sure 👍🏻

**Review by RavinduSachintha — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @devblin 
Thanks for your contribution, keep it up!

**Comment by Danushka96** [comment]
@all-contributors please add @devblin for code and doc

**Pull request merged**